src/material/dialog/dialog-ref.ts

```diff
@@ -108,25 +104,36 @@ export class MatDialogRef<T, R = any> {
       filter(event => event.phaseName === 'start'),
       take(1)
     )
-    .subscribe(event => {
+    .subscribe(() => {
       this._beforeClosed.next(dialogResult);
       this._beforeClosed.complete();
       this._state = MatDialogState.CLOSED;
       this._overlayRef.detachBackdrop();
-
-      // The logic that disposes of the overlay depends on the exit animation completing, however
-      // it isn't guaranteed if the parent view is destroyed while it's running. Add a fallback
-      // timeout which will clean everything up if the animation hasn't fired within the specified
-      // amount of time plus 100ms. We don't need to run this outside the NgZone, because for the
-      // vast majority of cases the timeout will have been cleared before it has the chance to fire.
-      this._closeFallbackTimeout = setTimeout(() => {
-        this._overlayRef.dispose();
-      }, event.totalTime + 100);
     });
 
     this._containerInstance._startExitAnimation();
     this._state = MatDialogState.CLOSING;
-  }
+
+    // The logic that disposes of the overlay depends on the exit animation completing, however
+    // it isn't guaranteed if the parent view is destroyed before that event. Add a fallback
+    // observer which will clean everything up if the animation hasn't completed within a specified
+    // amount of time plus 100ms. We don't need to run this outside the NgZone, because for the
```

Comment:
We can probably drop the last sentence here since we don't have a direct `setTimeout` anymore.

Reply:
Done